src/guide-rngs.md

```diff
@@ -6,8 +6,314 @@ however, often you can just use [`thread_rng`]. This function
+output. This requirement is formalized as the [next-bit test]; this is
+roughly stated as: given the first *k* bits of a random sequence, the
+sequence satisfies the next-bit test if there is no algorithm able to
+predict the next bit using reasonable computing power.
+
+A further security that *some* CSPRNGs provide is forward secrecy:
+in the event that the CSPRNGs state is revealed at some point, it must be
+infeasible to reconstruct previous states or output. Note that many CSPRNGs
+*do not* have forward secrecy in their usual formulations.
+
+As an outsider it is hard to get a good idea about the security of an
+algorithm. People in the field of cryptography spend a lot of effort
+analyzing existing designs, and what was once considered good may now turn
+out to be weaker. Generally it is best to use algorithms well-analyzed by
+experts, such as those recommended by NIST or ECRYPT.
+
+## State and seeding
+
+It is worth noting that a CSPRNG's security relies absolutely on being
+seeded with a secure random key. Should the key be known or guessable, all
+output of the CSPRNG is easy to guess. This implies that the seed should
+come from a trusted source; usually either the OS or another CSPRNG. Our
+seeding helper trait, [`FromEntropy`], and the source it uses
+([`EntropyRng`]), should be secure. Additionally, [`ThreadRng`] is a CSPRNG,
+thus it is acceptable to seed from this (although for security applications
+fresh/external entropy should be preferred).
+
+Further, it should be obvious that the internal state of a CSPRNG must be
+kept secret. With that in mind, our implementations do not provide direct
+access to most of their internal state, and `Debug` implementations do not
+print any internal state. This does not fully protect CSPRNG state; code
+within the same process may read this memory (and we allow cloning and
+serialisation of CSPRNGs for convenience). Further, a running process may be
+forked by the operating system, which may leave both processes with a copy
+of the same generator.
+
+## Not a crypto library
+
+It should be emphasised that this is not a cryptography library; although
+Rand does take some measures to provide secure random numbers, it does not
+necessarily take all recommended measures. Further, cryptographic processes
+such as encryption and authentication are complex and must be implemented
+very carefully to avoid flaws and resist known attacks. It is therefore
+recommended to use specialized libraries where possible, for example
+[openssl], [ring] and the [RustCrypto libraries].
+
+
+# Extra features
+
+Some PRNGs may provide extra features, like:
+
+- Support for multiple streams, which can help with parallel tasks.
+- The ability to jump or seek around in the random number stream;
+with large periood this can be used as an alternative to streams.
+
+
+# Further reading
+
+There is quite a lot that can be said about PRNGs. The [PCG paper] is a
+very approachable explaining more concepts.
```

Comment:
This sentence is missing a word?

Reply:
Or has too many? I think @pitdicker wrote that.